backend/src/main/java/heartbeat/service/report/GenerateReporterService.java

```diff
@@ -621,22 +628,49 @@ private List<PipelineCSVInfo> generateCSVForPipelineWithCodebase(CodebaseSetting
 	}
 
 	public InputStreamResource fetchCSVData(ExportCSVRequest request) {
-		deleteOldCSV();
-		return csvFileGenerator.getDataFromCSV(request.getDataType(), Long.parseLong(request.getCsvTimeStamp()));
+		long csvTimeStamp = Long.parseLong(request.getCsvTimeStamp());
+		validateExpireAndDeleteOldCSV(csvTimeStamp);
+		return csvFileGenerator.getDataFromCSV(request.getDataType(), csvTimeStamp);
 	}
 
-	private void deleteOldCSV() {
-		File directory = new File("./csv/");
-		File[] files = directory.listFiles();
+	private void validateExpireAndDeleteOldCSV(long csvTimeStamp) {
 		long currentTimeStamp = System.currentTimeMillis();
-		for (File file : files) {
-			String fileName = file.getName();
-			String[] splitResult = fileName.split("\\s*\\-|\\.\\s*");
-			String timeStamp = splitResult[1];
-			if (Long.parseLong(timeStamp) < currentTimeStamp - 36000000) {
-				file.delete();
+		if (validateExpire(currentTimeStamp, csvTimeStamp)) {
+			throw new NotFoundException("csv not found");
+		}
+		deleteOldCSV(currentTimeStamp, new File("./csv/"));
```

Comment:
Invalid logic! 
If validateExpire(currentTimeStamp, csvTimeStamp) is true, then it will throw exception and no delete happend.
if validateExpire(currentTimeStamp, csvTimeStamp) is false, then it will delete the file which will cause the downloading failed.

Btw, when delete specific file, we need to delete that file instead of the whole folder!